### ISSUE TITLE
[Enhancement]: Group payment methods from the same extension together in the editor sidebar

### ISSUE THREAD

**Issue by alexflorisca** [woothemes/woocommerce#50691]
### Describe the solution you'd like

In the checkout block, when you click on the Payment Options block, there is a list of active payment methods in the right sidebar. If there are multiple methods registered by the same extension, these should be grouped together. 

In the screenshot below, WooPayments has 2 registered payment methods. The second one (Bancontact) is all the way at the bottom. They should be grouped together.

<img width="1108" alt="Screenshot 2024-08-14 at 17 53 18" src="https://github.com/user-attachments/assets/97c9a356-3083-40bf-8ec1-0163aa6184b2">

**Comment by wavvves** [comment]
@alexflorisca the link icon for an empty entry hints at `type:bug` instead of enhancement 🤔

**Comment by alexflorisca** [comment]
Thanks for pointing that out. That's not what this issue is about, but it's definitely worth looking at that as well when working on this

**Comment by ralucaStan** [comment]
@alexflorisca is this just about the order of display or do you hope to achieve a visual grouping as well?

I see in your screen that there is an empty payment method. Was that a dummy payment you had set up?

**Comment by alexflorisca** [comment]
Yeah this is about visually grouping them together, so that all patyment options from the same plugins are displayed together. I can't remember what the empty payment was, but yeah you can ignore that

**Comment by pierorocca** [comment]
@yjailin, need your expertise here. Grouping at a minimum seems logical but I've never understood why the sidebar gets cluttered with a link per payment method. With 2-3 gateways installed this is a giant list of repetitive links.

**Comment by yjailin** [comment]
@pierorocca This is being addressed as part of the ongoing checkout sidebar redesign, which consolidates settings at the block level. The payment method list clutter will be resolved there, please have a look at pfHfG4-29C-p2. :)

**Comment by pierorocca** [comment]
Perfect. Closing this in favor of the future direction which consolidates everything down to a single "Manage payments" link.
